ui/src/app/base/reducers/pod/pod.ts

```diff
@@ -26,25 +10,128 @@ const initialState = {
   saved: false,
   saving: false,
   selected: [],
+  statuses: {},
 };
 
-const pod = createStandardReducer(podActions, initialState, {
-  REFRESH_POD_ERROR: (state: PodState, action: RefreshPodErrorAction) => {
-    state.errors = action.error;
-    state.saving = false;
-    state.saved = false;
+export const ACTIONS = [
+  {
+    status: "deleting",
+    type: "DELETE_POD",
   },
-  REFRESH_POD_SUCCESS: (state: PodState, action: RefreshPodSuccessAction) => {
-    for (const i in state.items) {
-      if (state.items[i].id === action.payload.id) {
-        state.items[i] = action.payload;
-        break;
-      }
-    }
+  {
+    status: "refreshing",
+    type: "REFRESH_POD",
   },
-  SET_SELECTED_PODS: (state: PodState, action: SelectPodAction) => {
-    state.selected = action.payload;
-  },
-});
+];
+
+export const DEFAULT_STATUSES = ACTIONS.reduce((statuses, action) => {
+  statuses[action.status] = false;
+  return statuses;
+}, {});
+
+const pod = createNextState((draft, action) => {
```

Comment:
As per your note, would be nice to be able to extend the standard reducer.

Reply:
Yeah for sure! For now I've mostly copied what was in the machines reducer, but there's a few differences with pods so I think it makes sense to do the refactor as the next step.